pkg/controller/operator/common/util.go

```diff
@@ -179,6 +179,28 @@ func VolumeRevisionLabelsModifierFactory(ctx context.Context, client ctrlruntime
 	}
 }
 
+// VersionLabelModifierFactory adds the version label for Deployments and their corresponding pods.
+func VersionLabelModifierFactory(version string) reconciling.ObjectModifier {
+	return func(create reconciling.ObjectReconciler) reconciling.ObjectReconciler {
+		return func(existing ctrlruntimeclient.Object) (ctrlruntimeclient.Object, error) {
+			obj, err := create(existing)
+			if err != nil {
+				return obj, err
+			}
+
+			deployment, ok := obj.(*appsv1.Deployment)
+			if !ok {
+				return obj, nil
```

Comment:
This should be an error maybe. Silently not doing anything if someone used the modifier on something invalid could lead to sadness.

For other modifiers, like the volume revision thingy, I think we also support DaemonSets and StatefulSets -- if we decide to skip those here for now, it would be good if the code screamed at us if someone decided to use this in the future on a DS/STS.

Reply:
that is a really good point. Added an error